tasks/eva_2779/correct_assembly_error_in_rat.py

```diff
@@ -0,0 +1,268 @@
+                    else:
+                        variant[key] = correction_map[key]
+                for inactive in variant['inactiveObjects']:
+                    inactive['hashedMessage'] = id_creation_func(inactive)
+                variant.pop('_id')
+                update_statements.append(pymongo.ReplaceOne(filter_dict, variant))
+            result_update = collection.with_options(write_concern=WriteConcern(w="majority", wtimeout=1200000)) \
+                .bulk_write(requests=update_statements, ordered=False)
+            total_updated += result_update.modified_count
+        logger.info(f'{total_updated} documents updated in {collection_name}')
+    except Exception as e:
+        print(traceback.format_exc())
+        raise e
+    return total_updated
+
+
+def replace_variant_entities(mongo_source, batch_size):
+    # submitted variants on MT chromosome
+    replace_document_with_correct_information(
+        mongo_source, "submittedVariantEntity", get_submitted_SHA1,
+        {'seq': 'GCA_015227675.1', 'contig': 'CM026996.1'},
+        {'seq': 'GCA_015227675.2', 'contig': 'AY172581.1'},
+        batch_size
+    )
+    replace_document_with_correct_information(
+        mongo_source, "dbsnpSubmittedVariantEntity", get_submitted_SHA1,
+        {'seq': 'GCA_015227675.1', 'contig': 'CM026996.1'},
+        {'seq': 'GCA_015227675.2', 'contig': 'AY172581.1'},
+        batch_size
+    )
+
+    # submitted variants Not on MT chromosome
+    replace_document_with_correct_information(
+        mongo_source, "submittedVariantEntity", get_submitted_SHA1, {'seq': 'GCA_015227675.1'},
+        {'seq': 'GCA_015227675.2'}, batch_size
+    )
+    replace_document_with_correct_information(
+        mongo_source, "dbsnpSubmittedVariantEntity", get_submitted_SHA1, {'seq': 'GCA_015227675.1'},
+        {'seq': 'GCA_015227675.2'}, batch_size
+    )
+
+    # Clustered variants on MT chromosome
+    replace_document_with_correct_information(
+        mongo_source, "clusteredVariantEntity", get_clustered_SHA1,
+        {'asm': 'GCA_015227675.1', 'contig': 'CM026996.1'},
+        {'asm': 'GCA_015227675.2', 'contig': 'AY172581.1'},
+        batch_size
+    )
+    replace_document_with_correct_information(
+        mongo_source, "dbsnpClusteredVariantEntity", get_clustered_SHA1,
+        {'asm': 'GCA_015227675.1', 'contig': 'CM026996.1'},
+        {'asm': 'GCA_015227675.2', 'contig': 'AY172581.1'},
+        batch_size
+    )
+
+    # Clustered variants Not on MT chromosome
+    replace_document_with_correct_information(
+        mongo_source, "clusteredVariantEntity", get_clustered_SHA1, {'asm': 'GCA_015227675.1'},
+        {'asm': 'GCA_015227675.2'}, batch_size
+    )
+    replace_document_with_correct_information(
+        mongo_source, "dbsnpClusteredVariantEntity", get_clustered_SHA1, {'asm': 'GCA_015227675.1'},
+        {'asm': 'GCA_015227675.2'}, batch_size
+    )
+
+    # MERGED Operations on MT chromosomes
```

Comment:
Hope you have ensured from the database that the event types laid out here is the exhaustive list..

Reply:
I check and the database only contains MERGED and RS_SPLIT but the next operation would apply to all types anyway.